### ISSUE TITLE
Adding reading sections to an assignment

### ISSUE THREAD

**Issue by bnmnetp** [RunestoneInteractive/RunestoneServer#957]
When I select a chapter/subchapter from the tree-view I expect that it would add something to the assignment table to indicate that reading the section is part of the assignment.  But it does not appear.

Also, we discussed that we need to have some kind of flag to indicate whether a particular question should be reproduced on the assignment page, or left to be done in-line.  I expect that would be a column in the assignment_questions table, but I don't see it there yet?  Or perhaps there is some other way to know.  This will be important for #955 to work properly.

**Comment by bjones1** [comment]
Fixed in https://github.com/RunestoneInteractive/RunestoneServer/tree/assignment_interface, please close.

**Comment by bnmnetp** [comment]
OK, just a few comments about some polish that I think may help, and some questions.

1.  the divisions between whole assignment, reading part, and exercise part are not clear. especially where the whole assignment and reading part come together.   -- I think a clear delineation and even a simple heading would help. 
2.  There is lots of blank space to the right where some simple help text would guide instructors through the process. -- Remember many of our users are high school teachers that are not developers.
1. I'm assuming the autograder dropdown refers to how the reading assignment is graded, but I couldn't explain how that works.
1. I would have expected that the in-reading components such as multiple choice etc would have appeared under the reading section, but I guess I have to add them in the next section?  I think this will be counter intuitive for many instructors that just want their students to read these sections and do the widgets in the section.   In fact to be honest I was surprised that there were two different tree views, one for reading stuff and one for exercise stuff.
1. I think the Write Assignment button can go now, as Hillary will have a working dynamic assignment page generator for us shortly.
1.  It looks like the tree views are being generated by querying chapters and subchapters of the base course rather than the specific course. This is fine for everyone except those that are doing highly customized books.

**Comment by bjones1** [comment]
Thanks, that's helpful. So I think:

- Items 1-3 are probably GUI improvements / explanatory text, correct? Perhaps item 4 as well?
- Item 5: Yay! I'll be happy to get rid of this gunk.
- Item 6: thanks for checking. I'll let @presnick fix this one.

**Comment by bnmnetp** [comment]
Thanks for the update @bjones1, I'm afraid we had a big misunderstanding about the **Write Assignment** button.

Regarding the readings part of the assignment.  Could either of you (@bjones1 or @presnick ) explain the difference between the autograding choices and the threshold values.  I swear I'm not trying to be dense but I don't understand what the choices on the dropdown mean in the context of these are the sections to read.  How do all of the embedded components figure into the grade?

<img width="509" alt="screen shot 2017-07-25 at 12 29 25 pm" src="https://user-images.githubusercontent.com/51115/28585246-4df472c0-7135-11e7-9cc0-72fbc01489bd.png">

For the problems section...

<img width="510" alt="screen shot 2017-07-25 at 12 35 32 pm" src="https://user-images.githubusercontent.com/51115/28585421-d83d95c4-7135-11e7-80ef-dd26457ad7d5.png">

My biggest question is should ALL of the problems checked, be included in the dynamically generated assignment page?  If there is a Check Your understanding multiple choice question that you would want a student to answer while doing the reading should that be part of the generated "worksheet" ?

We are so close to getting this thing really wrapped up!!

**Comment by presnick** [comment]
I promise I will write up something tonight or tomorrow morning on this.
There is a logic to all this, but it needs some user interface elements to
make it clearer to teachers (and programmers!)

**Comment by bjones1** [comment]
I'll defer all reading assignment explanations to @presnick, since I don't understand it that well either. I'm just a humble programmer... :)

I've put comments in #969 to resolve the misunderstanding -- let's continue that portion of the conversation there.

**Comment by presnick** [comment]
Based on the latest, two-column version of the assignments page that I see, it looks like everyone understands pretty well my vision of the two separate sections. I will try to edit the HTML to make the explanations even clearer to teachers, especially the idea that the readings segment specifies the section of the final assignment page that contains links to chapters/subchapters, and the Problems segment specifies the section of the final assignment page that contains problems right on the page, not links to any other pages.

I have added one new issue, suggesting that we have collapsible divs rather than two columns. https://github.com/RunestoneInteractive/RunestoneServer/issues/974

I have also added a new issue with a revised design for the readings section, which will resolve your question about the auto-grader thing, and make it more parallel with the table for the embedded problems segment. https://github.com/RunestoneInteractive/RunestoneServer/issues/975
